### PR TITLE
Retrieve cemeteries from CDW

### PR DESCRIPTION
**Pull request by jnguy-va**
Get cemeteries from CDW instead of ArcGIS.

Some other ArcGIS calls were removed due to the variables not being used.

https://vajira.max.gov/browse/API-1758

### CODE REVIEW

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollector.java [hunk 1]
```diff
@@ -1,44 +1,67 @@
...
-          .filter(c -> !equalsIgnoreCase(c.attributes().siteType(), "office"))
+      return requestCdwCemeteries().stream()
+          .filter(c -> !equalsIgnoreCase(c.siteType(), "office"))
           .map(
               facility ->
                   CemeteriesTransformer.builder()
-                      .arcgisFacility(facility)
-                      .csvWebsite(websites.get("nca_" + facility.attributes().siteId()))
+                      .cdwFacility(facility)
+                      .csvWebsite(websites.get("vba_" + facility.siteId()))
```
Comment (taylorrassmannlibertyits) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollector.java:64]: should be nca here

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/FacilitiesCollector.java [hunk 2]
```diff
@@ -87,7 +78,6 @@ public FacilitiesCollector(
           "Loading websites took {} millis for {} entries",
           totalWatch.stop().elapsed(TimeUnit.MILLISECONDS),
           websites.size());
-      checkState(!websites.isEmpty(), "No website entries");
```
Comment (taylorrassmannlibertyits) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/FacilitiesCollector.java:90]: why was this removed?

Comment (jnguy-va) [reply facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/FacilitiesCollector.java:90]: Adding back in, that was when my merge got a bit wonky.

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/FacilitiesCollector.java [hunk 3]
```diff
@@ -180,15 +170,14 @@ static String withTrailingSlash(@NonNull String url) {
...
             .websites(websites)
+            .jdbcTemplate(jdbcTemplate)
             .build()
             .collect();
 
     log.info(
-        "Collected: Health {},  Benefits {},  Vet centers {}, "
-            + "Non-national cemeteries {}, Cemeteries {}",
+        "Collected: Health {},  Benefits {},  Vet Centers {}, "
+            + "State Cemeteries {}, Cemeteries {}",
```
Comment (taylorrassmannlibertyits) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/FacilitiesCollector.java:181]: Non-National changed to state?

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/FacilitiesCollector.java [hunk 4]
```diff
@@ -242,10 +231,9 @@ static String withTrailingSlash(@NonNull String url) {
                 (rs, rowNum) -> toVastEntity(rs)));
 
     log.info(
-        "Loading VAST took {} millis for {} entries",
+        "Loading cemeteries took {} millis for {} entries",
```
Comment (taylorrassmannlibertyits) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/FacilitiesCollector.java:235]: keep VAST

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/FacilitiesCollector.java [hunk 5]
```diff
@@ -242,10 +231,9 @@ static String withTrailingSlash(@NonNull String url) {
...
         watch.stop().elapsed(TimeUnit.MILLISECONDS),
         entities.size());
-    checkState(!entities.isEmpty(), "No App.Vast entries");
```
Comment (taylorrassmannlibertyits) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/FacilitiesCollector.java:248]: don't remove?

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/FacilitiesCollector.java (hunk 3 above)
Comment (jnguy-va) [reply facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/FacilitiesCollector.java:181]: Should be `Non-national cemeteries` as per:

https://github.com/department-of-veterans-affairs/lighthouse-facilities/commit/0db5beb86b86657f04641d8ade7c08cad2406115

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CdwCemeteries.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+  String visitationHoursWeekend;
+
+  String websiteUrl;
+
+  BigDecimal longitude;
+
+  BigDecimal latitude;
+
+  static final class CdwCemeteriesBuilder {}
```
Comment (jblefkove-va) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CdwCemeteries.java:53]: This can be removed

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollector.java [hunk 7]
```diff
@@ -47,40 +70,40 @@
     }
   }
 
-  /** Requests ArcGIS VA_Cemeteries_Facilities in application/json. */
+  /** Requests CDW VA_Cemeteries_Facilities in application/json. */
```
Comment (jblefkove-va) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollector.java:73]: I don't know why this said `VA_Cemeteries_Facilities` before... just `cemeteries` is sufficient

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollector.java [hunk 8]
```diff
@@ -47,40 +70,40 @@
...
+                    + "LONGITUDE_DD,"
+                    + "Website_URL"
+                    + " FROM App.FacilityLocator_NCA",
+                (rs, rowNum) -> toCdwCemeteries(rs)));
     log.info(
-        "Loading cemeteries took {} millis for {} features",
+        "Loading cemeteries took {} millis for {} entries",
         totalWatch.stop().elapsed(TimeUnit.MILLISECONDS),
-        cemeteries.features().size());
-    checkState(!cemeteries.features().isEmpty(), "No ArcGIS cemeteries");
```
Comment (jblefkove-va) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollector.java:83]: restore this check

Comment (jnguy-va) [reply facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollector.java:83]: Restored below

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollector.java [hunk 9]
```diff
@@ -47,40 +71,41 @@
     }
   }
 
-  /** Requests ArcGIS VA_Cemeteries_Facilities in application/json. */
+  /** Requests CDW cemeteries in application/json. */
```
Comment (taylorrassmannlibertyits) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollector.java:74]: I don't think this is in application json anymore?

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollector.java [hunk 10]
```diff
@@ -47,40 +71,41 @@
...
+                (rs, rowNum) -> toCdwCemeteries(rs)));
     log.info(
-        "Loading cemeteries took {} millis for {} features",
+        "Loading cemeteries took {} millis for {} entries",
         totalWatch.stop().elapsed(TimeUnit.MILLISECONDS),
-        cemeteries.features().size());
-    checkState(!cemeteries.features().isEmpty(), "No ArcGIS cemeteries");
-    return cemeteries;
+        cdwCemeteries.size());
+    checkState(!cdwCemeteries.isEmpty(), "No CDW cemeteries");
```
Comment (taylorrassmannlibertyits) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollector.java:108]: Just checking here, does `checkState` evaluate the boolean and log if the evaluation is false?

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesTransformer.java [hunk 11]
```diff
@@ -12,47 +12,46 @@
...
-                        .zip(attributes.mailZip())
+                        .address1(checkAngleBracketNull(cdwFacility.mailAddress1()))
+                        .address2(checkAngleBracketNull(cdwFacility.mailAddress2()))
+                        .city(cdwFacility.mailCity())
+                        .state(upperCase(cdwFacility.mailState(), Locale.US))
+                        .zip(cdwFacility.mailZip())
                         .build())
                 .build())
-        .phone(phone(attributes.phone(), attributes.fax()))
+        .phone(phone(cdwFacility.phone(), cdwFacility.fax()))
```
Comment (taylorrassmannlibertyits) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesTransformer.java:45]: I think you need to do the additional phone number check here?

Comment (taylorrassmannlibertyits) [reply facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesTransformer.java:45]: For phone numbers that are 000-000-0000

File: facilities/src/test/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollectorTest.java [hunk 12]
```diff
@@ -3,48 +3,56 @@
...
+                    .mailAddress2("")
+                    .mailCity("Schuylerville")
+                    .mailState("NY")
+                    .mailZip("12871-1721")
+                    .phone("")
+                    .fax("5184630787")
+                    .visitationHoursWeekday("Sunrise - Sundown")
+                    .visitationHoursWeekend("Sunrise - Sundown")
+                    .latitude(new BigDecimal("42.703844900000036"))
+                    .longitude(new BigDecimal("-73.72356499999995"))
```
Comment (taylorrassmannlibertyits) [inline facilities/src/test/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollectorTest.java:49]: Do you want to have a website url?

File: facilities/src/test/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollectorTest.java [hunk 13]
```diff
@@ -64,15 +72,15 @@ void collect() {
                                     .physical(
                                         Facility.Address.builder()
                                             .address1("Cemetery Avenue")
-                                            .address2(null)
+                                            .address2("")
```
Comment (taylorrassmannlibertyits) [inline facilities/src/test/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollectorTest.java:75]: If you want to put some newer code through its paces you could have `address2` be `"<Null>"`

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollector.java (hunk 10 above)
Comment (jblefkove-va) [reply facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollector.java:108]: `checkState` verifies that the boolean expression is true. If it is not, it throws an `IllegalStateException` with the given message

Comment (jblefkove-va) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollector.java:108]: the equivalent message for benefits is `"No App.FacilityLocator_VBA entries"`

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesTransformer.java (hunk 11 above)
Comment (jblefkove-va) [reply facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesTransformer.java:45]: it's handled inside the `phone()` method

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollector.java [hunk 14]
```diff
@@ -5,40 +5,64 @@
...
 @Slf4j
 final class CemeteriesCollector {
-  @NonNull private final String arcgisUrl;
 
   @NonNull private final Map<String, String> websites;
 
-  @NonNull private final RestTemplate restTemplate;
+  private final JdbcTemplate jdbcTemplate;
+
+  /** Convert the results into a CdwBenefits Object. */
```
Comment (jblefkove-va) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollector.java:29]: copypasta

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CdwCemeteries.java [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+import java.math.BigDecimal;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Value;
+
+@Value
+@Builder
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class CdwCemeteries {
```
Comment (jblefkove-va) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CdwCemeteries.java:12]: How about `CdwCemetery` since it's a single entity?

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollector.java [hunk 16]
```diff
@@ -47,40 +71,41 @@
     }
   }
 
-  /** Requests ArcGIS VA_Cemeteries_Facilities in application/json. */
+  /** Requests CDW cemeteries in application/json. */
   @SneakyThrows
-  private ArcGisCemeteries requestArcGisCemeteries() {
+  private List<CdwCemeteries> requestCdwCemeteries() {
```
Comment (jblefkove-va) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollector.java:76]: `queryCdwCemeteries` would be a better name since this is now a database query instead of a web service call

File: facilities/src/test/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollectorTest.java [hunk 17]
```diff
@@ -3,48 +3,57 @@
...
+                    .mailCity("Schuylerville")
+                    .mailState("NY")
+                    .mailZip("12871-1721")
+                    .phone("")
+                    .fax("5184630787")
+                    .visitationHoursWeekday("Sunrise - Sundown")
+                    .visitationHoursWeekend("Sunrise - Sundown")
+                    .latitude(new BigDecimal("42.703844900000036"))
+                    .longitude(new BigDecimal("-73.72356499999995"))
+                    .websiteUrl("http://www.testme.com/")
```
Comment (jnguy-va) [inline facilities/src/test/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollectorTest.java:50]: Added in a URL for the test

File: facilities/src/test/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollectorTest.java [hunk 18]
```diff
@@ -3,48 +3,57 @@
...
-        .thenReturn(body);
+    when(jdbcTemplate.query(any(String.class), any(RowMapper.class)))
+        .thenReturn(
+            List.of(
+                CdwCemetery.builder()
+                    .siteId("088")
+                    .fullName("Albany Rural")
+                    .siteType("Rural")
+                    .siteAddress1("Cemetery Avenue")
+                    .siteAddress2("<Null>")
```
Comment (jnguy-va) [inline facilities/src/test/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollectorTest.java:35]: Test for "<Null>"

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollector.java [hunk 19]
```diff
@@ -26,10 +26,10 @@
 
   private final JdbcTemplate jdbcTemplate;
 
-  /** Convert the results into a CdwBenefits Object. */
+  /** Convert the results into a CdwCemeteries Object. */
```
Comment (taylorrassmannlibertyits) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollector.java:29]: Cemetery (not plural)

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollector.java [hunk 20]
```diff
@@ -26,7 +26,7 @@
 
   private final JdbcTemplate jdbcTemplate;
 
-  /** Convert the results into a CdwCemeteries Object. */
+  /** Convert the results into a CdwCemetery Object. */
   @SneakyThrows
   static CdwCemetery toCdwCemeteries(ResultSet rs) {
```
Comment (taylorrassmannlibertyits) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/collector/CemeteriesCollector.java:31]: I guess the method should be renamed too :)

**Review by taylorrassmannlibertyits — APPROVED** [review]

**Review by jblefkove-va — APPROVED** [review]
